### PR TITLE
feat(batch-exports): Timezone-aware schedules for batch exports

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

The temporal schedules that power batch exports are not aware of the team's timezone, so Temporal assumes they are UTC (Author's note: Oh boy, it's a timezone problem, buckle up).

This impacts several areas:
* All daily exports run at the same time (midnight UTC). This can lead to thundering herd problem, which we mitigate with a jitter, but ideally users would self-distribute better across the day already.
* Backfills are messy:
  * For some reason, the frontend assumes that dates inputted are in the user's local timezone **NOT** in their project's timezone.
  * If the user's local timezone doesn't match the schedule's timezone (which right now is only UTC), then the backfill will not match regular runs. For example, if a user in UTC+2 requests a backfill of a UTC export, then their backfill runs will cover periods bounded by dates like YYYY-MM-DDT22:00:00, as the UTC offset is "corrected".

## Changes

* Frontend calendar input now replaces the timezone of dates inputted by the user to the project's timezone. **NOTE**: We are **replacing** the timezone, **not converting** the date. This means that if a user in UTC+2 working on a UTC project inputs `YYYY-MM-DDT00:00:00+02:00` it will be replaced into `YYYY-MM-DDT00:00:00+00:00` (Notice the time component wasn't adjusted despite the timezone changing).
  * This solves the problem of date inputs being "corrected" when sent to the backend: In the backend, we know the timezone of the team, but we don't know the timezone of the user. So, we can convert dates to the team's timezone without fear of double-correcting.
* Future schedules are now timezone aware. Changing existing schedules would mean creating a gap or duplicates for all existing batch exports, so I opted for keeping existing batch exports in UTC and not updating them. 
  * TODO: Offer the user a path forward to updating the timezone of the batch export. This is also relevant if the user ever changes the project's timezone.
* Backfills are now aware of the schedule's timezone, and will adjust inputs accordingly. Essentially, the backfill workflow will convert the inputs to the schedules timezone, and afterwards convert them again to UTC as Temporal Workflow ID's have to be UTC (If the project's timezone and the schedule's timezone are both UTC, then we wouldn't need the double conversion, but leaving this minor optimization for later).
* Frontend calendar input now offers hour and minute precision for hourly and 5-minute batch exports.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yessir.

## How did you test this code?

Handful of tests added and updated.

### CODE REVIEW

File: frontend/src/scenes/pipeline/BatchExportBackfillModal.tsx [hunk 1]
```diff
@@ -54,18 +56,46 @@ export function BatchExportBackfillModal({ id }: BatchExportRunsLogicProps): JSX
...
+                    // project's timezone".
+                }
                 <LemonField name="start_at" label="Start Date" className="flex-1">
                     {({ value, onChange }) => (
-                        <LemonCalendarSelectInput value={value} onChange={onChange} placeholder="Select start date" />
+                        <LemonCalendarSelectInput
+                            value={value}
+                            onChange={(date) => {
+                                const projectDate = date.tz(timezone, true)
+                                onChange(projectDate)
```
Comment (tomasfarias) [inline frontend/src/scenes/pipeline/BatchExportBackfillModal.tsx:70]: @benjackwhite By any chance, do you have any opinion on this? I was quite surprised to learn we were not already assuming dates inputted here were in the project's timezone. I've made the change only in our modal, but I'm wondering if this is a problem everywhere, or if I am the one confused.

Comment (benjackwhite) [reply frontend/src/scenes/pipeline/BatchExportBackfillModal.tsx:70]: I've almost never used that component tbh. I would assume that it is rarely used for things as specific as batch exports is using it

Comment (benjackwhite) [reply frontend/src/scenes/pipeline/BatchExportBackfillModal.tsx:70]: For instance with most analytics queries the timezone is applied server side based on the team (which makes sense as it sends a non-timezone aware date usually.

**Review by benjackwhite — APPROVED** [review]
This makes sense. I also think it makes sense overall that this is not-timezone aware as a default. It's only something like this that requires a strict declaration of timezone (i.e. we dont want batch exports to change if you change your project timezone whereas graphs we do want to change)

**Comment by tomasfarias** [comment]
> This makes sense. I also think it makes sense overall that this is not-timezone aware as a default.

Do keep in mind that the component _is_ already timezone aware, just not a timezone we know of: It's assuming inputs are in the user's timezone, so definitely aware.

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**